beacon-chain/blockchain/service.go

```diff
@@ -234,16 +253,99 @@ func (c *ChainService) blockProcessing(processedBlock chan<- *types.Block) {
 	}
 }
 
+func (c *ChainService) blockProcessingNew(processedBlock chan<- *types.Block) {
+	subBlock := c.incomingBlockFeed.Subscribe(c.incomingBlockChan)
+	defer subBlock.Unsubscribe()
+	for {
+		select {
+		case <-c.ctx.Done():
+			log.Debug("Chain service context closed, exiting goroutine")
+			return
+
+		// Listen for a newly received incoming block from the feed. Blocks
+		// can be received either from the sync service, the RPC service,
+		// or via p2p.
+		case block := <-c.incomingBlockChan:
+			if c.currentSlot < block.SlotNumber() {
+				c.unProcessedBlocks[block.SlotNumber()] = block
+				continue
+			}
+
+			if c.currentSlot > block.SlotNumber() {
+				log.Debugf(
+					"Block slot number is lower than the currently processed slot in the beacon state %d",
+					block.SlotNumber())
+				continue
+			}
+
+			if err := c.processBlock(block); err != nil {
+				log.Error(err)
+				processedBlock <- nil
+				continue
+			}
+
+			// Push the block to trigger the fork choice rule.
+			processedBlock <- block
+		}
+	}
+}
+
+// This assigns the current slot of the
+// beacon node, functioning as the beacon node's
+// local clock.
+func (c *ChainService) slotTracker(slotTicker <-chan uint64, tickerDone func()) {
+
+	defer tickerDone()
+
+	for {
+		select {
+		case <-c.ctx.Done():
+			log.Debug("Chain service context closed, exiting goroutine")
+			return
+		case slot := <-slotTicker:
+			c.currentSlot = slot
+
+			c.checkForSkippedSlots()
+
+			beaconState, err := c.beaconDB.GetState()
+			if err != nil {
+				log.Debugf("Unable to retrieve beacon state %v", err)
+				continue
+			}
+
+			// Setting per slot counters
+
+			// Updating beacon state slot
+			beaconState.SetSlot(slot)
+			vreg := beaconState.ValidatorRegistry()
```

Comment:
I think L320 - L331 should be a block operation function in `core`
1.) get validator registry
2.) get proposer from validator registry
3.) update proposer's randao layer
4.) and then set it

Reply:
Agreed it might be better to push it to `core` instead of the service layer